host-configs/llnl/blueos_3_ppc64le_ib_p9/clang@upstream_link_with_nvcc.cmake

```diff
@@ -4,27 +4,30 @@
 # SPDX-License-Identifier: (BSD-3-Clause)
 
 #------------------------------------------------------------------------------
-# Example host-config file for the blue_os ray cluster at LLNL
+# Example host-config file for the blue_os cluster at LLNL
 #------------------------------------------------------------------------------
 #
 # This file provides CMake with paths / details for:
 #  C/C++, OpenMP, MPI, and Cuda
 # 
+# It demonstrates a more complex build, where the code is linked with nvcc.
+#------------------------------------------------------------------------------
+# Warning: This host-config does not currently work with CMake@3.13 or above
+#          due to a linking error. 
```

Comment:
Spent a while trying to track this one down, with help from @white238.

The problem appears to be related to new CMake functionality to prepend linker flags/paths (w/ `-Wl,-rpath,`). This does not play nicely with `nvcc`'s desired `Xlinker -rpath -Xlinker ` prefix.

@ajkunen suggested that it might be related to a dynamic library (`hwloc`) that we are using in the new mpi+cuda smoke test.

As of now, this host-config currently works with CMake@3.9 but not with CMake@3.14.

Reply:
@white238 fixed this in #308